features/step_definitions/scheduler.rb

```diff
@@ -24,5 +24,30 @@
     opts = {resource: 'scheduler', resource_name: name, p: patch_json, type: 'merge' }
     @result = _admin.cli_exec(:patch, **opts)
     raise "Cannot restore scheduler: #{name}" unless @result[:success]
+    timeout = 300
+    wait_for(timeout) do
+      @result = admin.cli_exec(:get, resource: "clusteroperators", resource_name: "kube-scheduler", o: "jsonpath={.status.conditions[?(.type == \"Progressing\")].status}")
+      if @result[:response] == "True"
+        break
+      end
+    end
+    wait_for(timeout) do
+      @result = admin.cli_exec(:get, resource: "clusteroperators", resource_name: "kube-scheduler", o: "jsonpath={.status.conditions[?(.type == \"Progressing\")].status}")
```

Comment:
@kasturinarra this line is the same as line 29.  Typo perhaps?

Reply:
@pruan-rht not a typo actually, it is the same line but checking for two different conditions, one for true and another for false.  Let me explain you here.

Once the scheduler cluster is patched it takes sometime for the kube-scheduler to come into progressing = True and if we do not wait for the kube-scheduler progressing status to be true  then script will move further and it will fail with out proper restoration of scheduler, that is the reason for having it twice.